pipeline/measuring.py

```diff
@@ -0,0 +1,307 @@
+import numpy as np
+import sqlalchemy as sa
+
+from scipy import signal
+
+from pipeline.parameters import Parameters
+from pipeline.data_store import DataStore
+from pipeline.utils import parse_session
+
+from models.base import SmartSession
+from models.cutouts import Cutouts
+from models.measurements import Measurements
+from models.enums_and_bitflags import BitFlagConverter
+
+from improc.photometry import iterative_photometry
+from improc.tools import make_gaussian
+
+
+class ParsMeasurer(Parameters):
+    def __init__(self, **kwargs):
+        super().__init__()
+
+        self.annulus_radii = self.add_par(
+            'annulus_radii',
+            [7.5, 10.0],
+            list,
+            'Inner and outer radii of the annulus. '
+        )
+
+        self.annulus_units = self.add_par(
+            'annulus_units',
+            'pixels',
+            str,
+            'Units for the annulus radii. Can be pixels or fwhm. '
+            'This describes the units of the annulus_radii input. '
+            'Use "pixels" to make a constant annulus, or "fwhm" to '
+            'adjust the annulus size for each image based on the PSF width. '
+        )
+
+        # TODO: should we choose the "best aperture" using the config, or should each Image have its own aperture?
+        self.chosen_aperture = self.add_par(
```

Comment:
This is how you specify which aperture to choose... but where is the list of apertures passed in?

OK, looks like it comes out of the zeropoint model.

Hmm.  We might want to have the abillity to do measurements in apertures other than the ones that were previously determined in the zeropoint model.  That does add complication, because we don't have a predetermined aperture correction, so we have to determine it on the fly.

Probably make an issue for this.

Reply:
See my comment above for the apertures coming from the ZP.